src/main/scala/mesosphere/marathon/tasks/TaskTracker.scala

```diff
@@ -1,45 +1,47 @@
 package mesosphere.marathon.tasks
 
-import java.io._
 import javax.inject.Inject
 
 import mesosphere.marathon.MarathonConf
 import mesosphere.marathon.Protos._
 import mesosphere.marathon.metrics.Metrics
-import mesosphere.marathon.state.{ PathId, StateMetrics, Timestamp }
-import mesosphere.util.state.{ PersistentEntity, PersistentStore }
+import mesosphere.marathon.state.{ EntityStore, MarathonTaskState, PathId, StateMetrics, Timestamp }
 import org.apache.mesos.Protos.{ TaskState, TaskStatus }
 import org.slf4j.LoggerFactory
 
-import scala.collection.JavaConverters._
 import scala.collection._
 import scala.collection.concurrent.TrieMap
 import scala.collection.immutable.Set
 import scala.concurrent.{ Await, Future }
 
 class TaskTracker @Inject() (
-    store: PersistentStore,
+    store: EntityStore[MarathonTaskState],
```

Comment:
Please create a separate TaskRepository instead of using the entity store directly.